pkg/gui/controllers/helpers/worktree_helper.go

```diff
@@ -0,0 +1,80 @@
+package helpers
+
+import (
+	"fmt"
+	"github.com/go-errors/errors"
+	"github.com/jesseduffield/lazygit/pkg/commands"
+	"github.com/jesseduffield/lazygit/pkg/commands/models"
+	"github.com/jesseduffield/lazygit/pkg/gui/context"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	"io/fs"
+	"log"
+	"os"
+)
+
+type IWorktreeHelper interface {
+	GetMainWorktreeName() string
+	GetCurrentWorktreeName() string
+}
+
+type WorktreeHelper struct {
+	c        *types.HelperCommon
+	git      *commands.GitCommand
+	contexts *context.ContextTree
+	model    *types.Model
+}
+
+func NewWorktreeHelper(
+	c *types.HelperCommon,
+	git *commands.GitCommand,
+	contexts *context.ContextTree,
+	model *types.Model) *WorktreeHelper {
+	return &WorktreeHelper{
+		c:        c,
+		git:      git,
+		contexts: contexts,
+		model:    model,
+	}
+}
+
+func (self *WorktreeHelper) GetMainWorktreeName() string {
+	for _, worktree := range self.model.Worktrees {
+		if worktree.Main() {
+			return worktree.Name()
+		}
+	}
+
+	return ""
+}
+
+func (self *WorktreeHelper) IsCurrentWorktree(w *models.Worktree) bool {
+	pwd, err := os.Getwd()
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	return pwd == w.Path
+}
+
+func (self *WorktreeHelper) IsWorktreePathMissing(w *models.Worktree) bool {
+	if _, err := os.Stat(w.Path); err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return true
+		}
+		log.Fatalln(fmt.Errorf("failed to check if worktree path `%s` exists\n%w", w.Path, err).Error())
+	}
+	return false
+}
+
+func (self *WorktreeHelper) NewWorktree() error {
+	return self.c.Prompt(types.PromptOpts{
+		Title: self.c.Tr.NewWorktreePath,
+		HandleConfirm: func(response string) error {
+			self.c.LogAction(self.c.Tr.Actions.CreateWorktree)
```

Comment:
Can we wrap this in a loader, given it takes a couple seconds for me. Have a look at `WithWaitingStatus`